vchan/libvchan.h

```diff
@@ -47,6 +47,18 @@ libvchan_t *libvchan_server_init(int domain, int port, size_t read_min, size_t w
 
 libvchan_t *libvchan_client_init(int domain, int port);
 
+/* An alternative path for client connection:
+ * 1. Call libvchan_client_init_async first.
+ * 2. Call libvchan_client_init_fd_for_select() and wait for the FD to be readable.
+ * 3. When readable, call libvchan_client_init_async_finish().
+ *
+ * Repeat steps 2-3 until libvchan_client_init_async_finish returns 0. Abort on
+ * negative values (error).
+ */
+libvchan_t *libvchan_client_init_async(int domain, int port);
+int libvchan_client_init_fd_for_select(libvchan_t *ctrl);
```

Comment:
Should `select` still be included in the names, given that it is obsolete on \*nix?  Also, there needs to be an API that can be used on Windows.

Reply:
Windows version uses `HANDLE` in this place (see `windows/include/libvchan.h`). But I'm not going to implement Windows version right now, maybe will leave a TODO comment there.